### PR TITLE
C#/Java: Add some model generator summary debugging queries.

### PR DESCRIPTION
**Pull request by michaelnebel**
In this PR we add a couple of debugging queries related to model generation for C# and Java.
- A query for returning flow paths (if we need to figure out why the model generator generates a certain model).
- A query for partial flow (if we need to figure out why the model generator doesn't generate a certain model).

### CODE REVIEW

File: csharp/ql/src/utils/modelgenerator/debug/CaptureSummaryModelsPartialPath.ql [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+module PartialFlow = PropagateFlow::FlowExplorationFwd<explorationLimit/0>;
+
+from
+  PartialFlow::PartialPathNode source, PartialFlow::PartialPathNode sink,
+  DataFlowSummaryTargetApi api, DataFlow::ParameterNode p
+where
+  PartialFlow::partialFlow(source, sink, _) and
+  p = source.getNode() and
+  p.asParameter() = api.getParameter(0)
```
Comment (hvitved) [inline csharp/ql/src/utils/modelgenerator/debug/CaptureSummaryModelsPartialPath.ql:25]: Should this be restricted to only the first parameter?

File: java/ql/src/utils/modelgenerator/debug/CaptureSummaryModelsPartialPath.ql [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+module PartialFlow = PropagateFlow::FlowExplorationFwd<explorationLimit/0>;
+
+from
+  PartialFlow::PartialPathNode source, PartialFlow::PartialPathNode sink,
+  DataFlowSummaryTargetApi api, DataFlow::ParameterNode p
+where
+  PartialFlow::partialFlow(source, sink, _) and
+  p = source.getNode() and
+  p.asParameter() = api.getParameter(0)
```
Comment (hvitved) [inline java/ql/src/utils/modelgenerator/debug/CaptureSummaryModelsPartialPath.ql:26]: Same.

**Review by hvitved — APPROVED** [review]
LGTM. Just one question.

File: csharp/ql/src/utils/modelgenerator/debug/CaptureSummaryModelsPartialPath.ql (hunk 1 above)
Comment (michaelnebel) [reply csharp/ql/src/utils/modelgenerator/debug/CaptureSummaryModelsPartialPath.ql:25]: No - I will change that.

**Review by hvitved — APPROVED** [review]

**Pull request merged**